model/dependencies.go

```diff
@@ -42,14 +43,22 @@ func (di *dependencyIncluder) include(initialDeps []TVPair) ([]TVPair, error) {
 			outPairs = append(outPairs, pair)
 		}
 	}
+	// we deactivate variants that do not have specific activation set in the
+	// generated project but have been generated as dependencies of other
+	// explicitly inactive variants
+	for variant, addToActivationInfo := range di.deactivateGeneratedVariantTasks {
```

Comment:
Why add these here as opposed to activationVariants directly? Is it because we could be setting it back to false in handle?

Reply:
Yes - we need to wait for the recursion of `handle` to complete since even a single active task that depends on the newly generated dependency should make cause the generated dependency to also be active